hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ServiceInternalProcessingTimeEnricher.java

```diff
@@ -0,0 +1,73 @@
+package org.hypertrace.traceenricher.enrichment.enrichers;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+import org.hypertrace.core.datamodel.ApiNodeEventEdge;
+import org.hypertrace.core.datamodel.Event;
+import org.hypertrace.core.datamodel.MetricValue;
+import org.hypertrace.core.datamodel.StructuredTrace;
+import org.hypertrace.core.datamodel.shared.ApiNode;
+import org.hypertrace.core.datamodel.shared.trace.AttributeValueCreator;
+import org.hypertrace.traceenricher.enrichedspan.constants.EnrichedSpanConstants;
+import org.hypertrace.traceenricher.enrichment.AbstractTraceEnricher;
+import org.hypertrace.traceenricher.trace.util.ApiTraceGraph;
+import org.hypertrace.traceenricher.trace.util.ApiTraceGraphBuilder;
+import org.slf4j.LoggerFactory;
+
+public class ServiceInternalProcessingTimeEnricher extends AbstractTraceEnricher {
+
+  private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(
+      ServiceInternalProcessingTimeEnricher.class);
+
+  public void enrichTrace(StructuredTrace trace) {
+    ApiTraceGraph apiTraceGraph = ApiTraceGraphBuilder.buildGraph(trace);
+    List<ApiNode<Event>> apiNodeList = apiTraceGraph.getApiNodeList();
+    for (ApiNode<Event> apiNode : apiNodeList) {
+      List<ApiNodeEventEdge> edges = apiTraceGraph.getOutboundEdgesForApiNode(apiNode);
+      int edgeDurationSum = 0;
```

Comment:
nit: long

Reply:
Why? Will this sum ever exceed 2,147,483,647ms? That's around 25 days.